Gardener Local installation doesn't recognize kind

**What happened**:
I ran the following commands:
```
kind create cluster --name test-cluster
export KUBECONFIG="$(kind get kubeconfig-path --name="test-cluster")"
make dev-setup
```
got:
```
Creating cluster "test-cluster" ...
 ✓ Ensuring node image (kindest/node:v1.15.0) 🖼
 ✓ Preparing nodes 📦
 ✓ Creating kubeadm config 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Cluster creation complete. You can now use the cluster with:
export KUBECONFIG="$(kind get kubeconfig-path --name="test-cluster")"
kubectl cluster-info

make[1]: Entering directory '/home/travis/gopath/src/github.com/gardener'
Unsupported Kubernetes Environment: 
Makefile:30: recipe for target 'dev-setup' failed
make[1]: *** [dev-setup] Error 1
make[1]: Leaving directory '/home/travis/gopath/src/github.com/gardener'
```

**What you expected to happen**:
```
Creating cluster "test-cluster" ...
 ✓ Ensuring node image (kindest/node:v1.15.0) 🖼
 ✓ Preparing nodes 📦
 ✓ Creating kubeadm config 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Cluster creation complete. You can now use the cluster with:
export KUBECONFIG="$(kind get kubeconfig-path --name="test-cluster")"
kubectl cluster-info

Found kind ...
namespace/garden created
namespace/garden-dev created
deployment.apps/etcd created
service/etcd created
service/gardener-apiserver created
service/gardener-controller-manager created
endpoints/gardener-apiserver created
endpoints/gardener-controller-manager created
apiservice.apiregistration.k8s.io/v1beta1.garden.sapcloud.io created
apiservice.apiregistration.k8s.io/v1alpha1.core.gardener.cloud created
validatingwebhookconfiguration.admissionregistration.k8s.io/validate-namespace-deletion created
make[1]: Leaving directory '/home/travis/gopath/src/github.com/gardener'
```
**How to reproduce it (as minimally and precisely as possible)**:
see in "I ran the following commands"

**Anything else we need to know?**:
I know what the problem is, see below

**Environment**:
- Travis-CI on GCP
- kind stable
- kubectl stable
- Gardener local master

**Problem in Code**:
see: https://github.com/gardener/gardener/blob/master/hack/common#L31

**Reason**:
when running  ```kind create cluster --name test-cluster``` the cluster name is appended to the beginning of the returned answer from the command.
but when running ```kind create cluster`` it defaults to kind.

**Proof**:
when running:
```
kind create cluster
export KUBECONFIG="$(kind get kubeconfig-path)"
echo "$(kubectl get node -o jsonpath="{.items[0].metadata.name}")"
```
you get: 
```kind-control-plane```

and when running:
```
kind create cluster --name test-cluster
export KUBECONFIG="$(kind get kubeconfig-path --name="test-cluster")"
echo "$(kubectl get node -o jsonpath="{.items[0].metadata.name}")"
```
you get:
```test-cluster-control-plane```

**Additional problem**:
https://github.com/gardener/gardener/blob/master/hack/common#L25
doesn't allow to workaround the issue and override ```__k8s_nodename``` int he environment using export.

--- a/docs/development/local_setup.md
+++ b/docs/development/local_setup.md
@@ -199,7 +199,7 @@ $ sed -i -e 's/Logging: true/Logging: false/g' dev/20-componentconfig-gardener-c
 
 The Gardener exposes the API servers of Shoot clusters via Kubernetes services of type `LoadBalancer`. In order to establish stable endpoints (robust against changes of the load balancer address), it creates DNS records pointing to these load balancer addresses. They are used internally and by all cluster components to communicate.
 You need to have control over a domain (or subdomain) for which these records will be created.
-Please provide an *internal domain secret* (see [this](../../example/10-secret-internal-domain.yaml) for an example) which contains credentials with the proper privileges. Further information can be found [here](../deployment/configuration.md).
+Please provide an *internal domain secret* (see [this](../../example/10-secret-internal-domain.yaml) for an example) which contains credentials with the proper privileges. Further information can be found [here](../concepts/configuration.md).
 
 ```bash
 $ kubectl apply -f example/10-secret-internal-domain-unmanaged.yaml

--- a/hack/common
+++ b/hack/common
@@ -19,6 +19,7 @@ set -e
 DOCKER_FOR_DESKTOP="docker-for-desktop"
 MINIKUBE="minikube"
 KIND="kind"
+KIND_NODE_NAME_SUFFIX="-control-plane"
 
 API_SERVER_SECURE_PORT="${API_SERVER_SECURE_PORT:-8443}"
 
@@ -37,7 +38,7 @@ k8s_env() {
         echo "$MINIKUBE"
         return
     fi
-    if [[ "$node_name" == "$KIND"* ]]; then
+    if [[ "$node_name" == *"$KIND_NODE_NAME_SUFFIX" ]]; then
         echo "$KIND"
         return
     fi

